turnitintooltwo_view.class.php

```diff
@@ -1295,7 +1295,7 @@ public function get_submission_inbox_row($cm, $turnitintooltwoassignment, $parts
                             .html_writer::tag('span', "/".$parts[$partid]->maxmarks,
                                     array("class" => "grademark_grade"));
                 } else if ($turnitintooltwoassignment->turnitintooltwo->gradedisplay == 1) { // 1 is percentage.
-                    $submissiongrade = round($submissiongrade / $parts[$partid]->maxmarks * 100, 1).'%';
+                    $submissiongrade = ($submissiongrade == '--') ? $submissiongrade : round($submissiongrade / $parts[$partid]->maxmarks * 100, 1).'%';
```

Comment:
would it be better to use https://www.php.net/manual/en/function.is-numeric.php

Reply:
ahh I think `is_numeric` would be safer. Prior to this, [there is a check](https://github.com/ixiongtii/moodle-mod_turnitintooltwo/blob/fix/grade_display_setting_unable_to_load_php8/turnitintooltwo_view.class.php#L1268-L1270) if the submission grade is null and it will set it to '--' if so. So, I think it'll always either be '--' or a string numeric and php can interpret numeric strings as ints and still do basic operands with them.

But I would think that `is_numeric` would be safer to use just in case